src/services/notification/mod.rs

```
@@ -40,6 +41,19 @@ pub trait Notifier {
 	/// # Returns
 	/// * `Result<(), anyhow::Error>` - Success or error
 	async fn notify(&self, message: &str) -> Result<(), anyhow::Error>;
+	/// Sends a notification with a custom JSON payload
+	///
+	/// # Arguments
+	/// * `payload` - The Object payload to send
+	///
+	/// # Returns
+	/// * `Result<(), anyhow::Error>` - Success or error
+	async fn notify_with_payload<T: Serialize + ?Sized + Send + Sync>(
+		&self,
+		_payload: &T,
+	) -> Result<(), anyhow::Error> {
+		Ok(())
```

Do you think this is a good approach or should we return an error saying that if you need to use this method you have to override it?

---

Yes :) let's do a "not implemented" error